### ISSUE TITLE
Device update error

### ISSUE THREAD

**Issue by filipSu** [RonnyWinkler/homey.mqtthub#6]
Hi,

Once device is created, I can't seem to update topics definition in settings. Every time I change something in the topic json, I get an error. The change I wanted to make was only to change the display name on one capability, so I haven't messed up the json. There seems to be a problem with `settings` local API call - it returns 500. I've noticed that whole `capabilities` property is missing in the API request payload.

![Screenshot 2024-12-19 at 14 44 06](https://github.com/user-attachments/assets/ff011268-9e0b-4ef0-be0b-60bab5f31972)
![Screenshot 2024-12-19 at 14 44 18](https://github.com/user-attachments/assets/f6984013-27f4-42f6-bf58-1b697188958a)
![Screenshot 2024-12-19 at 14 44 34](https://github.com/user-attachments/assets/16f3310b-c28f-485f-a7a1-f2699f04904a)
![Screenshot 2024-12-19 at 14 44 42](https://github.com/user-attachments/assets/dd7e7581-a152-4835-b1f4-877f5872f71b)

It doesn't happen on all devices, but it always does happen on `battery`

MQTT Hub version: `4.3.1 Test`
Homie version: `12.3.0-rc.7`
Device class: `battery`
Topic setting:
```
{
  "measure_battery": {
    "capability": "measure_battery",
    "stateTopic": "evcc/site/batterySoc",
    "setTopic": "",
    "valueTemplate": "",
    "outputTemplate": "",
    "displayName": "Batteryy"
  },
  "battery_charging_state": {
    "capability": "battery_charging_state",
    "stateTopic": "evcc/site/batteryPower",
    "setTopic": "",
    "valueTemplate": "",
    "outputTemplate": "",
    "displayName": "Battery charging state"
  },
  "measure_power": {
    "capability": "measure_power",
    "stateTopic": "evcc/site/batteryPower",
    "setTopic": "",
    "valueTemplate": "",
    "outputTemplate": "",
    "displayName": "Power"
  }
}
```

**Comment by RonnyWinkler** [comment]
Hi, 
based on the code, this error can appear if the MQTT connection is not available on settings change (topic subscription). In this case, the settings are put to old values overwriting an JSON object that seem sto be readonly (perhaps changed with a Homey FW).
I changed it to use a copy instead. But I can't test if I don't get MQTT errors.

Please check the new test version 4.3.2

**Comment by filipSu** [comment]
Thanks for the answer @RonnyWinkler . I've tried it and it seems that this problem is fixed 💪 I can update the json from the UI, so you can close this issue 🙇‍♂️

Unfortunately there is a new issue with the cumulative flag 😅 I've posted on the [community board](https://community.homey.app/t/app-pro-mqtt-hub-community-version/113463/137). Should I create a new issue for it or do you want to discuss it in on the community board?

**Comment by filipSu** [comment]
Also I think there is a problem [here](https://github.com/RonnyWinkler/homey.mqtthub/commit/30b31875d5753461b84011a56ce4c3b8c6d88012#diff-2bf94fb61282f08a9334399f5a06e3ce18bdfd180430f12c9d7b75872a0abe74R40). You forgot to set `class` property in the new settings. Maybe this can be a problem for Homey 🤷‍♂️ I have to set it again after the device is created in the device settings (it is empty after device creation)

**Comment by filipSu** [comment]
resolved is 4.3.2. And [cumulative flag mentioned in earlier](https://community.homey.app/t/app-pro-mqtt-hub-community-version/113463/139) comment resolved in 4.3.3
